kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -510,25 +534,35 @@ def kube_job_failed(self, metric, scraper_config):
 
     def kube_job_status_failed(self, metric, scraper_config):
         for sample in metric.samples:
+            job_ts = 0
             tags = [] + scraper_config['custom_tags']
             for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
                 if label_name == 'job' or label_name == 'job_name':
                     trimmed_job = self._trim_job_tag(label_value)
+                    job_ts = self._extract_job_timestamp(label_value)
                     tags.append(self._format_tag(label_name, trimmed_job, scraper_config))
                 else:
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
-            self.job_failed_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
+            if job_ts != 0 and job_ts > self.failed_job_counts[frozenset(tags)].previous_run_max_ts:
+                self.failed_job_counts[frozenset(tags)].count += sample[self.SAMPLE_VALUE]
+                if job_ts > self.failed_job_counts[frozenset(tags)].current_run_max_ts:
+                    self.failed_job_counts[frozenset(tags)].current_run_max_ts = job_ts
```

Comment:
Is there an issue with `self.failed_job_counts` and `self. succeeded_job_counts ` growing in size ? Or the size is relatively fixed ?

Reply:
Yeah we wanted to expire jobs from these dicts, but from inside this check we can't tell if a job we don't see in the prometheus output won't trigger again (kube-state-metrics could have removed it because it ran more than an hour ago, but it's just scheduled to run once a day, in which case removing it will bring the counter back to zero).